### PR TITLE
Add simple mixture initializer.

### PR DESCRIPTION
**Pull request by MTCam**
This is part 1 of several, where the Pyro EOS and mixture PR's are getting pulled apart into more manageable chunks.

This one intends to add a simple mixture initialize.

### CODE REVIEW

File: mirgecom/initializers.py [hunk 1]
```diff
@@ -807,3 +808,77 @@ def exact_rhs(self, discr, q, t=0.0):
...
+            self, *, dim=3, nspecies=0,
+            pressure=101500.0, temperature=300.0,
+            massfractions=None, velocity=None,
+    ):
+        r"""Initialize mixture parameters.
+
+        Parameters
+        ----------
+        dim: int
+            specifies the number of dimensions for the lump
```
Comment (inducer) [inline mirgecom/initializers.py:830]: > lump?

File: mirgecom/initializers.py [hunk 2]
```diff
@@ -807,3 +808,77 @@ def exact_rhs(self, discr, q, t=0.0):
 
         return join_conserved(dim=self._dim, mass=massrhs, energy=energyrhs,
                               momentum=momrhs, species_mass=yrhs)
+
+
+class MixtureInitializer:
+    r"""Solution initializer for multi-species mixture.
```
Comment (inducer) [inline mirgecom/initializers.py:814]: What distinguishes this from, e.g. the `MulticomponentLump`?

File: mirgecom/initializers.py [hunk 3]
```diff
@@ -9,6 +9,7 @@
 .. autoclass:: MulticomponentLump
 .. autoclass:: Uniform
 .. autoclass:: AcousticPulse
+.. autoclass:: MixtureInitializer
 .. automethod: _make_pulse
 .. automethod: _make_uniform_flow
```
Comment (inducer) [inline mirgecom/initializers.py:14]: Sorry, unrelated, but only seeing this now: underscored functions should by definition not be part of a documented interface. Either un-underscore or remove from the docs.

File: mirgecom/initializers.py (hunk 2 above)
Comment (inducer) [inline mirgecom/initializers.py:814]: If this is uniform, would it be possible to generalize the existing `Uniform` initializer?

Comment (MTCam) [reply mirgecom/initializers.py:814]: 6132569

Comment (MTCam) [reply mirgecom/initializers.py:814]: I added some docs that will hopefully help make it more apparent how this init differs from the others.  The main thing is mixture-compatible EOS. 

(6132569)

File: mirgecom/initializers.py (hunk 3 above)
Comment (inducer) [reply mirgecom/initializers.py:14]: #230

**Review by inducer — APPROVED** [review]

File: mirgecom/initializers.py (hunk 3 above)
Comment (MTCam) [reply mirgecom/initializers.py:14]: Yes! #231

**Pull request merged**